src/Entity/User.php

```diff
@@ -37,67 +37,49 @@ class User implements UserInterface, EntityInterface
     /** @var string */
     private $username;
 
+    /** @var string|null */
+    private $plainPassword;
+
     /** @var string */
     private $password;
 
-    /** @var ArrayCollection|Assignment[] */
-    private $assignments;
-
     /** @var string[] */
-    private $roles = [];
-
-    /** @var string|null */
-    private $plainPassword;
+    private $roles;
 
     /** @var string|null */
     private $groupId;
 
-    public function __construct()
-    {
-        $this->assignments = new ArrayCollection();
-    }
-
-    public function getId(): UuidV6
-    {
-        return $this->id;
-    }
+    /** @var ArrayCollection|Assignment[] */
+    private $assignments;
 
-    public function setId(UuidV6 $id): self
-    {
+    public function __construct(
+        UuidV6 $id,
+        string $username,
+        string $plainPassword,
```

Comment:
should be:
```
string $plainPassword = null
```
right?

Reply:
No, because a user is not valid without a password. Also the event listener (UserPasswordEncoderListener) expects it to encode it.

Reply:
But do we have it when fetching from DB?